test/cases/schema-title-as-type-name.ts

```diff
@@ -0,0 +1,41 @@
+export var schema = {
+    "$schema": "http://json-schema.org/draft-03/schema",
+    "id": "http://mycompany.com/api/referencing.json",
+    "title": "Referencing",
+    "type": "object",
+    "properties": {
+        "ref": {
+            "$ref": "../../test/resources/ReferencedTypeWithId.json"
```

Comment:
No need for this 2-file test case. Just have a single file with a schema that has both a `title` and and `id`, and test that `title` takes precedence.